src/components/AppSelector/AppSelector.tsx

```diff
@@ -1,118 +1,74 @@
-import React from 'react'
-import { useParams, useHistory, generatePath, useRouteMatch } from 'react-router'
-import { mapByKey, useAsync } from '../common'
-import Select, { components } from 'react-select';
-import { ReactComponent as DropDownIcon } from '../../assets/icons/ic-chevron-down.svg';
-import StateManager, { ControlProps } from 'react-select';
+import React from 'react';
+import { showError } from '../common';
+import AsyncSelect from 'react-select/async';
+import { ServerErrors } from '../../modals/commonTypes';
+import { getAppList } from '../app/service';
+import { appSelectorStyle, DropdownIndicator } from './AppSelectorUtil';
 
-const AppSelector: React.FC<{
-    primaryKey: string; //url match
-    primaryValue: string;
-    matchedKeys: string[];
-    api: () => Promise<any>;
-    apiPrimaryKey?: string; //primary key to generate map
-    onChange?: ({ label, value }) => void;
-    formatOptionLabel?: ({ label, value, ...rest }) => React.ReactNode;
-    filterOption?: (option: any, searchString: string) => boolean;
-}> = ({ primaryKey, primaryValue, matchedKeys, api, apiPrimaryKey, onChange, formatOptionLabel = null, filterOption = null }) => {
-    const [loading, result, error, reload] = useAsync(api, []);
-    const listMap = mapByKey(result?.result || [], apiPrimaryKey || primaryKey);
-    const { path } = useRouteMatch();
-    const params = useParams();
-    const { push } = useHistory();
-    function selectApp(selected) {
-        if (onChange) {
-            onChange(selected)
-            return
+interface AppSelectorType {
+    onChange: ({ label, value }) => void;
+    appId: number;
+    appName: string;
+}
+export default function AppSelector({ onChange, appId, appName }: AppSelectorType) {
+    const defaultOption = [{ value: appId, label: appName }];
+    let selectedValue = defaultOption[0];
+    let timeoutId;
+    function noOptionsMessage(inputObj: { inputValue: string }): string {
```

Comment:
This function seems to be defined in 2 places. 
Can you please write it at a common place and use it at both places?